src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+
+import {
+  useState,
+  useContext,
+  useRef,
+  useEffect,
+  useCallback,
+} from 'react';
+
+import { TodosContext } from '../TodoContext/TodoContext';
+import { ErrorMessage } from '../../types/Error';
+import { USER_ID } from '../../utils/UserID';
+
+export const TodoHeader = () => {
+  const [newTitle, setNewTitle] = useState('');
+  const {
+    addCurrentTodo,
+    setErrorMessage,
+    isLoading,
+    setIsLoading,
+    todos,
+    updateCurrentTodo,
+  } = useContext(TodosContext);
+
+  const titleFocus = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (titleFocus.current && !isLoading) {
+      titleFocus.current.focus();
+    }
+  }, [isLoading]);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setErrorMessage(ErrorMessage.None);
+
+    if (!newTitle.trim()) {
+      setErrorMessage(ErrorMessage.EmptyTitle);
+
+      return;
+    }
+
+    setIsLoading(true);
+
+    addCurrentTodo({
+      title: newTitle.trim(),
+      completed: false,
+      userId: USER_ID,
+    })
+      .then(() => setNewTitle(''))
+      .catch(() => setIsLoading(false))
+      .finally(() => setIsLoading(false));
+  };
+
+  const toggleAllTodos = useCallback(() => {
+    const isEveryCompleted = todos.every(todo => todo.completed);
+
+    todos.forEach(todo => updateCurrentTodo(
+      { ...todo, completed: !isEveryCompleted },
+    ));
+  }, [todos]);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        type="button"
+        className="todoapp__toggle-all active"
```

Comment:
you should change className of this button depends on completed/active todos. check demo version